lib/tests/streamlit/scriptrunner/script_runner_test.py

```diff
@@ -445,6 +457,17 @@ def test_invalidating_cache(self):
             ],
         )
 
+    def test_memory_gc(self):
+        """Tests that we correctly handle scripts with runtime errors."""
+        usage_before = resource.getrusage(resource.RUSAGE_SELF)
+        scriptrunner = TestScriptRunner("memory_leak.py")
+        scriptrunner.enqueue_rerun()
+        scriptrunner.start()
+        scriptrunner.join()
+        usage_after = resource.getrusage(resource.RUSAGE_SELF)
+
+        self._assert_no_exceptions(scriptrunner)
```

Comment:
Curious about this one here. Is there a purpose to it other than checking that running the test does not produce errors?

Reply:
It is in progress. Sorry, that didn't leave a comment.